### PR TITLE
qvariant: add support for more types to QVariant

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Related to #11

Change-Id: Ic6aea2ad5f8265eafcf30bbab7aaf9e3eee77478

### CODE REVIEW

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
I think we should be able to remove a lot of code duplication before we merge this.

File: cxx-qt-lib/src/qvariant.rs [hunk 1]
```diff
@@ -253,6 +332,116 @@ impl Variant {
         }
     }
 
+    pub fn from_qcolor(color: Color) -> Self {
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply cxx-qt-lib/src/qvariant.rs]: Got this working with macros, i've left most of the tests as `Variant::from(value)` but `value.into() should also work.

File: cxx-qt-lib/src/qvariant.rs [hunk 2]
```diff
@@ -48,6 +63,34 @@ extern "C" {
     fn qvariant_init_from_i16(ptr: &mut MaybeUninit<cxx::UniquePtr<QVariant>>, i: i16);
     #[link_name = "cxxqt1$qvariant$init$from$i32"]
     fn qvariant_init_from_i32(ptr: &mut MaybeUninit<cxx::UniquePtr<QVariant>>, i: i32);
+    #[link_name = "cxxqt1$qvariant$init$from$qcolor"]
+    fn qvariant_init_from_qcolor(this: &mut MaybeUninit<cxx::UniquePtr<QVariant>>, color: &QColor);
```
[the remark this answers is not in the document]
Comment (LeonMatthesKDAB) [reply cxx-qt-lib/src/qvariant.rs:67]: Yes, if it adds additional dependencies and is non-trivial, skip this for now.

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM
I especially like the combination of the IntoQVariant trait with a macro to reduce code duplication even further :ok_hand:

**Pull request merged**